app/hooks/usePlannerBg.ts

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from "react";
+import { useWindowSize } from "react-use";
+
+type PlannerBgParams = {
+  bgWidth: number;
+  bgHeight: number;
+  pointOffsetX: number;
+  pointOffsetY: number;
+};
+
+// Dynamic background size. See this issue for more info: https://github.com/Sendouc/sendou.ink/issues/1161
+const bgSizeFactor = 0.8;
+
+export function usePlannerBg() {
+  const [plannerBgParams, setPlannerBgParams] = useState({});
```

Comment:
I think instead of doing type assertion below we should do:

```suggestion
  const [plannerBgParams, setPlannerBgParams] = useState<PlannerBgParams>({});
```

and give it some reasonable default value. Otherwise the return value of this function is kind of cheating. Or we can ofc:

```suggestion
  const [plannerBgParams, setPlannerBgParams] = useState<PlannerBgParams | null>(null);
```

And let caller handle it.

Reply:
Thanks Sendou! I knew I was missing something when it came down to the typing for the React Hook.

I went with your first suggestion :) - you can have another look & merge when ready. Glad to see that I did something significant for the map planner!